### PR TITLE
delete function added

### PR DESCRIPTION
**Pull request by Mike-Kirichenko**

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
+
+       
+        li.append(input);
 
+        doneDiv.setAttribute("class", "done");
+        doneDiv.innerHTML = ` <i class="fa fa-check-circle"></i>`;
+
+        li.append(doneDiv);
+
+        spanEdit.addEventListener("click", ()=>this.delete(toDo.id));
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 2]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
+        spanDelete.addEventListener("click", ()=>this.delete(toDo.id));
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 3]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
+        
+        this.listEl.append(li);
     }
-    delete(){
+    edit(){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 4]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
 
+    }
+    delete(id){
+        let toDoList = this.getData();
+        const newToDos = toDoList.filter(item=>item.id!==id?true:false);
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 5]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
+        spanDelete.addEventListener("click", ()=>this.delete(toDo.id));
+        
+        this.listEl.append(li);
     }
-    delete(){
+    edit(){
 
+    }
+    delete(id){
+        let toDoList = this.getData();
```
Comment (usatovaDasha) [inline js/script.js]: why do I need 2 identical strings 110 and 113?

File: js/script.js [hunk 6]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
+        const newToDos = toDoList.filter(item=>item.id!==id?true:false);
+        localStorage.setItem("toDoList", JSON.stringify(newToDos));
+        toDoList = this.getData();
+        if(toDoList.length < 1 ) {
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 7]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
-                    </div>
-                <input class="to-do-text" value="${toDo.name}" disabled>
-                    <div class="done">
-                        <i class="fa fa-check-circle"></i>
-                    </div>
-                </li>`;
-        });
-        return list;
+        this.listEl.innerHTML = "";
+        todos.forEach(toDo=>this.createTodo(toDo));
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 8]
```diff
@@ -60,44 +60,80 @@ class ToDoList{
...
     use(){
-        const toDoList = this.getData();
-        if(toDoList) this.listEl.innerHTML = this.listToHtml(toDoList);
+        const toDoList = this.getData();//retrive initial data
+
+        if(toDoList) this.listToHtml(toDoList);//render initial data
         else this.notify("To-do list is empty");
-        this.addButton.onclick = ()=>this.add();
+
+        this.addButton.onclick = () =>this.add();//add todo
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js (hunk 5 above)
Comment (M-kirichenko) [reply js/script.js:110]: to retrive data from localStorage again

Comment (Mike-Kirichenko) [reply js/script.js]: to retrieve data again from localstorage

Comment (usatovaDasha) [reply js/script.js]: so what does it have to do with re-receiving?

File: js/script.js [hunk 9]
```diff
@@ -11,6 +11,7 @@ class ToDoList{
         if(localStorage.toDoList) return JSON.parse(localStorage.toDoList);
         return false;
     }
+
     validate(name){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 10]
```diff
@@ -11,6 +11,7 @@ class ToDoList{
...
         let answer = true;
         if(name.length >= 5){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 11]
```diff
@@ -20,12 +21,13 @@ class ToDoList{
                     if(toDoList[i].name === name){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 12]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
         return answer;
     }
     notify(msg, elementRef){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 13]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
...
-        elementRef?elementRef.style.cssText = "border-bottom: 1px solid red":false;
+        elementRef ? elementRef.style.cssText = "border-bottom: 1px solid red" : false;
         this.msgEl.innerText = msg;
     }
     add(){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 14]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
...
         let toDoList = this.getData();
         const toDo = {};
+
         if(toDoList){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 15]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
...
             const lastId = toDoList[toDoList.length - 1].id;
             toDo.id = lastId + 1;
             toDo.name = this.inputText.value;
             toDo.done = false;
         }
+
         else{
```
Comment (usatovaDasha) [inline js/script.js]: code style

File: js/script.js [hunk 16]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
...
             toDoList = [];
             toDo.id = 1;
             toDo.name = this.inputText.value;
             toDo.done = false;
         }
+
         const validToDo = this.validate(toDo.name);
 
         if(validToDo === true) {
```
Comment (usatovaDasha) [inline js/script.js]: return the object with the desired flag and message, if required, but it's like a crutch

File: js/script.js [hunk 17]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
...
             toDoList.push(toDo);
             localStorage.setItem("toDoList", JSON.stringify(toDoList));
             this.inputText.value = "";
             this.msgEl.innerText = "";
-            this.inputText.style.cssText = "border-bottom: 1p solid black";
+            this.inputText.style.cssText = "border-bottom: 1px solid black";
             this.listToHtml(toDoList);
         }
+
         else this.notify(validToDo.msg, this.inputText);
```
Comment (usatovaDasha) [inline js/script.js]: code style

File: js/script.js [hunk 18]
```diff
@@ -34,36 +36,42 @@ class ToDoList{
...
     }
+
     createTodo(toDo){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js (hunk 5 above)
Comment (usatovaDasha) [reply js/script.js]: i don't see changes

File: js/script.js (hunk 6 above)
Comment (usatovaDasha) [reply js/script.js]: !

File: js/script.js [hunk 19]
```diff
@@ -108,22 +116,26 @@ class ToDoList{
...
+        
         if(toDoList.length < 1 ) {
             localStorage.removeItem("toDoList");
             this.notify("To-do list is empty");
         }
+
         this.listToHtml(toDoList);
     }
+
     listToHtml(todos){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 20]
```diff
@@ -108,22 +116,26 @@ class ToDoList{
...
         this.listEl.innerHTML = "";
         todos.forEach(toDo => this.createTodo(toDo));
     }
+
     use(){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 21]
```diff
@@ -108,8 +108,9 @@ class ToDoList{
 
   delete(id){
     const toDoList = this.getData();
-    const newToDos = toDoList.filter( item => item.id !== id ? true : false );
     
+    const newToDos = toDoList.filter( item => item.id !== id ? true : false );
```
Comment (usatovaDasha) [inline js/script.js]: const newToDos = toDoList.filter( item => item.id !== id);

File: js/script.js [hunk 22]
```diff
@@ -1,146 +1,138 @@
...
-            this.msgEl.innerText = "";
-            this.inputText.style.cssText = "border-bottom: 1px solid black";
-            this.listToHtml(toDoList);
-        }
-
-        else this.notify(validToDo.msg, this.inputText);
+    return answer;
+  }
+
+  notify(msg, elementRef){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 23]
```diff
@@ -1,146 +1,138 @@
...
+    elementRef ? elementRef.style.cssText = "border-bottom: 1px solid red" : false;
+    this.msgEl.innerText = msg;
+  }
+
+  add(){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 24]
```diff
@@ -1,146 +1,138 @@
...
+    let toDoList = this.getData();
+    const toDo = {};
+
+    if(toDoList){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 25]
```diff
@@ -1,146 +1,138 @@
...
+    li.append(navDiv);
+    input.setAttribute("class", "to-do-text");
+    input.setAttribute("value", toDo.name);
+    input.setAttribute("disabled", true);
+    li.append(input);
+    doneDiv.setAttribute("class", "done");
+    doneDiv.innerHTML = ` <i class="fa fa-check-circle"></i>`;
+    li.append(doneDiv);
+
+    spanEdit.addEventListener("click", () => this.delete(toDo.id));
```
Comment (usatovaDasha) [inline js/script.js]: 98 and 100 are the same functions?

File: js/script.js [hunk 26]
```diff
@@ -1,146 +1,138 @@
...
 
-        li.append(navDiv);
+  edit() {
 
-        input.setAttribute("class", "to-do-text");
-        input.setAttribute("value", toDo.name);
-        input.setAttribute("disabled", true);
+  }
 
+  delete(id){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

File: js/script.js [hunk 27]
```diff
@@ -1,146 +1,138 @@
...
+    const toDoList = this.getData();
+    const newToDos = toDoList.filter( item => item.id !== id ? true : false );
```
Comment (usatovaDasha) [inline js/script.js]: wrote error

File: js/script.js [hunk 28]
```diff
@@ -1,146 +1,138 @@
...
+  }
 
-        localStorage.setItem("toDoList", JSON.stringify(newToDos));
-        toDoList = this.getData();
-        
-        if(toDoList.length < 1 ) {
-            localStorage.removeItem("toDoList");
-            this.notify("To-do list is empty");
-        }
+  listToHtml(todos){
```
Comment (usatovaDasha) [inline js/script.js]: check the spaces

**Pull request merged**